### DISCUSSION TITLE
TiVo - An alternate Remote on iPhone

### DISCUSSION THREAD

**Discussion by tom76017** [GerryDazoo/Slinger#30] [Show and tell]
I found a free app in the Apple App Store that works pretty good for my TiVo Mini.  It's easy to setup and is fairly responsive given the Slingbox buffering.    I have a fixed IP address on my TiVo and have activated "Network Remote Control" on the TiVo.  The Back button is the older Zoom button.

**_"Peanut Remote for TiVo DVR"_**

**Comment by rblancarte** [comment]
I love the Tivo remote app. The bummer is that it doesn't allow you to remote control the Tivo when away from home. You have to be on the local network to do it.

**Reply by tom76017** [reply]
Mine works fine externally.

1. On your TiVo, Allow Network Control.
2. On your router, open port 31339 TCP and point it to your TiVo box
	Give your TiVo and Slinger servers static IP addresses.
3. On the Intranet, get a domain name for DDNS.
4. On your router, setup DDNS or use your ISP IP address.
	I use GoogleDomain ($12/yr) for registering a name and getting a DNS listing.
	or, get a static external IP address.
5. On the Peanut app, manual enter the Domain or external IP address

My local router lets me access both internal and external on the DNS name/port.

**Reply by rblancarte** [reply]
Just wanted to come back to this with my update - doesn't work on my Tivo Mini I have for my Slingbox. They don't support network remotes.

**Reply by tom76017** [reply]
![image](https://user-images.githubusercontent.com/26171888/197203559-617e047c-56e1-492e-8864-f4bb3b9b44ac.png)

**Reply by rblancarte** [reply]
Active and works when I am on my internal network. I can't change channels remotely (while I can on my 2 bolts)

**Reply by DVoorhees1957** [reply]
A few quick related questions. I'm trying to make the minimal number of changes to my existing network config. I've already enabled/allowed network-based remote controls on my TiVo.

Regarding your step #2: I thought this app communicated directly with my TiVo, not the Slingbox or the Slingbox server. Why would I need static IPs for anything except my TiVo?

Regarding your steps # 3 and 4: Do I really need DDNS or a domain name? Can't I just use the external/public IP address of my ISP's router?

**Reply by tom76017** [reply]
#2 - I use static addresses for each device.  Slinger and VLC need it for your Slingbox, and the Network Remote needs it for your TiVo.
#3 - For external access, you need a DDNS.  Most ISP's dynamically assign an IP to your modem.  This IP will change.  There are several free services.  example: https://www.noip.com/

**Comment by tom76017** [comment]
You have a firewall and routing issue on your router.  Nothing to do with Slingbox or Slinger.  I recall several years ago having an issue with Comcast not permitting me to access my LAN if I was on my LAN.  So I had to use my local IP's if I was on my LAN.  Then change it to my domain name with proper ip/port references on the router.

To add another TiVo, just use a different external port and internal IP with the same internal port (31399)

![image](https://user-images.githubusercontent.com/26171888/197339334-ed6df4e1-a7fd-41e2-a452-fa7f16f02624.png)

**Comment by DVoorhees1957** [comment]
I was hopeful that this approach might work for me, but I can only get it to work when my iPhone is on the local network with the TiVo. When I disable wifi on my phone, and try to set up the app to use the external IP of my router, and port 31339, it keeps trying to connect but can't. I have a port forwarding rule set up on my router (port 31339 TCP maps to 31339 of the desired TiVo), and I have network-based remote control enabled for this TiVo. Why isn't this working?

FOLLOW-UP EDIT
Never mind! I fat-fingered the port specification in the port forwarding rule, telling it 31399 instead of 31339. I changed the port forwarding rule, and it works fine now!

**Reply by tom76017** [reply]
You're not alone.   There's lots of things to do wrong, but once you get it working and comfortable with using another device it is better than the original Desktop software.  It's just another remote for me Samsung/VLC TV app.

**Comment by DVoorhees1957** [comment]
I just ran into an issue with this app. It seemed to work fine when I was testing with one of my TiVos (a Tivo Premier XL Series4, model 748000), but I just tried it with a different TiVo (a Tivo Premier, model 746500) and neither the guide button nor the channel up/channel down buttons seem to trigger the proper operation. Is there any way to check the key codes that are assigned to specific buttons - and more importantly, can those key codes be changed if necessary?

**Reply by GerryDazoo** [reply]
The short answer is yes. Plese read the release notes.

**Reply by DVoorhees1957** [reply]
Sorry if my comments were misleading, I was talking about the iPhone app called "Peanut Remote for Tivo DVR", which is what the author's original post was about. I can't find any info about customizing the buttons on the "Peanut Remote for Tivo DVR" in the release notes; are they there, and I just don't see them?

**Reply by GerryDazoo** [reply]
Sorry. I should have read the while thread. I was referring to the Slinger remote control definition file.

**Reply by DVoorhees1957** [reply]
Thanks, I figured. I may have to go back to using Slinger-based remote commands if I can't make this iPhone app work for both my TiVos.

I think you mentioned, Gerry, that you also use an iPhone app for sending remote control commands. If so, is it the Peanut Remote app that tom76017 mentioned at the top of this thread, or is it some other app? I still like the idea of having an iPhone-based remote control app, rather than a web browser based one...

**Reply by JoeM2016** [reply]
I think the peanut app is directed to the Ip address of the Tivo, you would have to change the address to the Tivo you want to control.

**Reply by GerryDazoo** [reply]
I don't have an iPhone. Any ol' browser can be used to send remote control commands.

**Reply by tom76017** [reply]
The TiVo has to be activated to permit "**_Network Remote Control_**".  The Peanut app and the Windows app have nothing to do with Slinger directly.  I can use the iPhone app and Windows app on my Edge, and Mini A92 and A95.  They send a code to the TiVo immediately.  **_Zoom_** is the only thing I don't use.

After looking at the TiVo site, it suggests that you reboot the TiVo after setting the Control.  I'm not sure if the Premier will work.

**Reply by DVoorhees1957** [reply]
JoeM2016, I know how to direct the output to 2 different TiVos, I've already done that (and can switch between the 2 as necessary). The problem is that, while the guide button and channel up/down buttons work entirely as expected with the Tivo Premier XL Series4 (model 748000), they do **NOT** work properly for the Tivo Premier (model 746500). Hence my question about whether the key codes assigned to those buttons could be changed/reassigned...

**Reply by DVoorhees1957** [reply]
Thanks for the reply, Tom, I have Network Remote Control set up already on both TiVos. The Peanut Remote app works just fine with the 1st TiVo. I know that Network Remote Control is working on the 2nd TiVo, because I could invoke TiVo commands remotely for that TiVo - just not the commands I intended (for instance, when I pressed the Peanut Remote app key to change channels, the TiVo instead displayed the current output format - 720p).

I suspect, but can't yet prove, that the issue has something to do with what TiVo calls "remote addresses". If you have multiple TiVos in the same room, and multiple TiVo remotes, you have to assign a different remote address to each remote, and pair it with the respective TiVo, so that commands from one remote don't affect both TiVos. Specifically, my guess is that the remote address that's implicitly or explicitly embedded in the Peanut Remote app is compatible with the remote address for my 1st TiVo, but not my 2nd. Or, I could be completely off base about the remote address thing - but it's the only explanation I can come up with right now that is consistent with the results I'm seeing.

**Reply by tom76017** [reply]
My TiVos are in separate rooms.  Your requirement sounds complicated.  I found this:

https://tivoidp.tivo.com/tivoCommunitySupport/s/article/How-to-Use-a-Control4-System-with-TiVo-DVRs

![image](https://user-images.githubusercontent.com/26171888/200946550-ca34a636-0b86-411d-a5c7-77490c6ec8b1.png)
